docs/debug.md

```diff
@@ -0,0 +1,82 @@
+<!--
+---
+linkTitle: "Debug"
+weight: 11
+---
+-->
+# Debug
+
+- [Overview](#overview)
+- [Debugging TaskRuns](#debugging-taskruns)
+  - [Adding Breakpoints](#adding-breakpoints)
+    - [Breakpoint on Failure](#breakpoint-on-failure)
+      - [Failure of a Step](#failure-of-a-step)
+      - [Halting a Step on failure](#halting-a-step-on-failure)
+      - [Exiting breakpoint](#exiting-breakpoint)
+- [Debug Environment](#debug-environment)
+  - [Mounts](#mounts)
+  - [Debug Scripts](#debug-scripts)
+
+
+## Overview
+
+`Debug` spec is used for troubleshooting and breakpointing runtime resources. This doc helps understand the inner 
+workings of debug in Tekton. Currently only the `TaskRun` resource is supported. 
+
+## Debugging TaskRuns
+
+The following provides explanation on how Debugging TaskRuns is possible through Tekton. To understand how to use 
```

Comment:
This seems like it's developer-focused to me. Wonder if it should live under `docs/developers/debug.md`?